### PR TITLE
feat(ci): Add a new metric to compute duration of required jobs

### PR DESCRIPTION
**Pull request by chouetz**
### What does this PR do?
Add a metric computing required jobs duration

### Motivation
It's not easy to compute this duration directly on ci-visibility (as jobs have dependency we must take into account). It will give us a precise information of the minimum CI requested time duration until further updates on the pipeline constraints

### Additional Notes
Sorry for the formatting, shipped with the ruff update..

### CODE REVIEW

**Review by FlorentClarret — COMMENTED** [review]
Left one question 🙂

**Review by CelianR — APPROVED** [review]

File: tasks/libs/pipeline/stats.py [hunk 1]
```diff
@@ -38,3 +130,40 @@ def get_failed_jobs_stats(project_name, pipeline_id):
...
+    series = []
+    max_duration, status = get_max_duration(project_name=project_name)
+    pipeline_tags = [
+        "repository:datadog-agent",
+        f"git_ref:{os.getenv('CI_COMMIT_REF_NAME')}",
+        f"status:{status}",
+    ]
+    series.append(
+        create_gauge(
+            metric_name="datadog.ci.required_duration",
```
[the remark this answers is not in the document]
Comment (chouetz) [reply tasks/libs/pipeline/stats.py:146]: Agreed

File: tasks/libs/pipeline/stats.py [hunk 2]
```diff
@@ -1,8 +1,100 @@
+import os
+import traceback
 from collections import Counter
+from datetime import datetime
 
+from invoke import Exit
+
+from tasks.libs.ciproviders.gitlab import Gitlab, get_gitlab_token
+from tasks.libs.common.datadog_api import create_count, create_gauge
```
[the remark this answers is not in the document]
Comment (chouetz) [reply tasks/libs/pipeline/stats.py:9]: `stats` module is currently only imported on the `notify` one, which was already importing theses, so I'm not sure it changes. However if we continue offloading code from the tasks modules, we should reduce the number of imports on top level and reduce the dependencies, wdyt?

**Pull request merged**